src/devices/wifi/nm-iwd-manager.c

```diff
@@ -307,24 +332,105 @@ interface_added (GDBusObjectManager *object_manager, GDBusObject *object,
                  GDBusInterface *interface, gpointer user_data)
 {
 	NMIwdManager *self = user_data;
+	NMIwdManagerPrivate *priv = NM_IWD_MANAGER_GET_PRIVATE (self);
+	GDBusProxy *proxy;
+
+	if (!priv->running)
+		return;
 
-	set_device_dbus_object (self, interface, object);
+	g_return_if_fail (G_IS_DBUS_PROXY (interface));
+
+	proxy = G_DBUS_PROXY (interface);
+
+	if (!strcmp (g_dbus_proxy_get_interface_name (proxy),
+	             NM_IWD_DEVICE_INTERFACE)) {
```

Comment:
Let's cache the interface-name in a variable, and use `nm_streq*()` instead of `strcmp()`?
```C
    interface = g_dbus_proxy_get_interface_name (proxy);
    if (nm_streq0 (interface, NM_IWD_DEVICE_INTERFACE)) {
        ...
```

Reply:
Ok, makes sense.